src/databricks/labs/ucx/hive_metastore/pipelines_migrate.py

```diff
@@ -0,0 +1,160 @@
+import logging
+from dataclasses import dataclass
+from functools import partial
+from typing import BinaryIO
+from collections.abc import Generator
+
+from databricks.labs.blueprint.parallel import Threads
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound, DatabricksError
+from databricks.labs.blueprint.installation import Installation
+
+from databricks.labs.ucx.assessment.pipelines import PipelinesCrawler, PipelineInfo
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class PipelineRule:
+    workspace_name: str
+    src_pipeline_id: str
+    target_catalog_name: str | None = None
+    target_schema_name: str | None = None
+    target_pipeline_name: str | None = None
+
+    @classmethod
+    def from_src_dst(
```

Comment:
This method is the same as the regular initialization, why introduce it?

Reply:
Did you resolve this comment? It does not look like it

Reply:
Yes, the pipelinemapping functionality was removed, but the leftover code was not cleaned up. Cleaned it up now.